pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
It is not done everywhere.  It is only done during the final display to the user and it doesn't apply only to errors returned from a ClientConfig object.

The display of errors is distinct from the errors themselves.  Mutating the actual errors reporting by failing components makes it impossible for anyone higher in the call stack to interpret failing errors.  Now they have to understand not only the error reported by our API, but also the different set of errors that are created to represent them.

If you want to display certain errors differently, then do that at the point of display by calling the transformation there.  That way all the code only has to know about the API errors themselves, not the second set of potential transformations.  Think of trying to handle it as you would translation.  You don't translate early, you translate just before display.